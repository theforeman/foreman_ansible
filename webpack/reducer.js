import { combineReducers } from 'redux';

import ansibleRolesSwitcher from './components/AnsibleRolesSwitcher/AnsibleRolesSwitcherReducer';

export default combineReducers({
  ansibleRolesSwitcher,
});
