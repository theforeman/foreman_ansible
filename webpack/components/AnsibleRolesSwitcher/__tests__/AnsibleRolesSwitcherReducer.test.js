import { testReducerSnapshotWithFixtures } from '@theforeman/test';

import reducer, { initialState } from '../AnsibleRolesSwitcherReducer';
import { ansibleRolesLong } from '../__fixtures__/ansibleRolesData.fixtures';

import {
  successPayload,
  successState,
  errorPayload,
} from '../__fixtures__/ansibleRolesSwitcherReducer.fixtures';

import {
  ANSIBLE_ROLES_REQUEST,
  ANSIBLE_ROLES_SUCCESS,
  ANSIBLE_ROLES_FAILURE,
  ANSIBLE_ROLES_ADD,
  ANSIBLE_ROLES_REMOVE,
  ANSIBLE_ROLES_ASSIGNED_PAGE_CHANGE,
} from '../AnsibleRolesSwitcherConstants';

const fixtures = {
  'should return initial state': {
    state: initialState,
    action: {
      type: undefined,
      payload: {},
    },
  },
  'should start loading on Ansible roles request': {
    state: initialState,
    action: {
      type: ANSIBLE_ROLES_REQUEST,
    },
  },
  'should stop loading on Ansible roles success': {
    state: initialState.set('loading', true),
    action: {
      type: ANSIBLE_ROLES_SUCCESS,
      payload: successPayload,
    },
  },
  'should stop loading on Ansible roles error': {
    state: initialState.set('loading', true),
    action: {
      type: ANSIBLE_ROLES_FAILURE,
      payload: { error: errorPayload },
    },
  },
  'should add Ansible role to assigned': {
    state: successState,
    action: {
      type: ANSIBLE_ROLES_ADD,
      payload: { role: ansibleRolesLong[8] },
    },
  },
  'should remove Ansible role from assigned': {
    state: successState,
    action: {
      type: ANSIBLE_ROLES_REMOVE,
      payload: { role: ansibleRolesLong[5] },
    },
  },
  'should change pagination for assigned roles': {
    state: successState,
    action: {
      type: ANSIBLE_ROLES_ASSIGNED_PAGE_CHANGE,
      payload: { pagination: { page: 20, perPage: 5 } },
    },
  },
};

describe('AnsibleRolesSwitcherReducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
