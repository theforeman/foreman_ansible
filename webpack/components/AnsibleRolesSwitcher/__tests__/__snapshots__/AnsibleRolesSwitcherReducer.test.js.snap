// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnsibleRolesSwitcherReducer should add Ansible role to assigned 1`] = `
Object {
  "assignedPagination": Object {
    "page": 1,
    "perPage": 20,
  },
  "assignedRoles": Array [
    Object {
      "id": 4,
      "inherited": true,
      "name": "geerlingguy.java",
    },
    Object {
      "id": 5,
      "name": "naftulikay.golang",
    },
    Object {
      "id": 6,
      "name": "theforeman.foreman_scap_client",
    },
    Object {
      "id": 9,
      "name": "chef.chef",
    },
  ],
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "itemCount": 10,
  "loading": false,
  "pagination": Object {
    "page": 1,
    "perPage": 5,
  },
  "results": Array [
    Object {
      "id": 1,
      "name": "sthirugn.motd",
    },
    Object {
      "id": 2,
      "name": "jtyr.ntp",
    },
    Object {
      "id": 3,
      "name": "rvm.ruby",
    },
    Object {
      "id": 4,
      "name": "geerlingguy.java",
    },
    Object {
      "id": 5,
      "name": "naftulikay.golang",
    },
    Object {
      "id": 6,
      "name": "theforeman.foreman_scap_client",
    },
    Object {
      "id": 7,
      "name": "ansible.ansible",
    },
    Object {
      "id": 8,
      "name": "puppet.puppet",
    },
    Object {
      "id": 9,
      "name": "chef.chef",
    },
    Object {
      "id": 10,
      "name": "salt.salt",
    },
    Object {
      "id": 11,
      "name": "anonymous.something",
    },
  ],
}
`;

exports[`AnsibleRolesSwitcherReducer should change pagination for assigned roles 1`] = `
Object {
  "assignedPagination": Object {
    "page": 20,
    "perPage": 5,
  },
  "assignedRoles": Array [
    Object {
      "id": 4,
      "inherited": true,
      "name": "geerlingguy.java",
    },
    Object {
      "id": 5,
      "name": "naftulikay.golang",
    },
    Object {
      "id": 6,
      "name": "theforeman.foreman_scap_client",
    },
  ],
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "itemCount": 11,
  "loading": false,
  "pagination": Object {
    "page": 1,
    "perPage": 5,
  },
  "results": Array [
    Object {
      "id": 1,
      "name": "sthirugn.motd",
    },
    Object {
      "id": 2,
      "name": "jtyr.ntp",
    },
    Object {
      "id": 3,
      "name": "rvm.ruby",
    },
    Object {
      "id": 4,
      "name": "geerlingguy.java",
    },
    Object {
      "id": 5,
      "name": "naftulikay.golang",
    },
    Object {
      "id": 6,
      "name": "theforeman.foreman_scap_client",
    },
    Object {
      "id": 7,
      "name": "ansible.ansible",
    },
    Object {
      "id": 8,
      "name": "puppet.puppet",
    },
    Object {
      "id": 9,
      "name": "chef.chef",
    },
    Object {
      "id": 10,
      "name": "salt.salt",
    },
    Object {
      "id": 11,
      "name": "anonymous.something",
    },
  ],
}
`;

exports[`AnsibleRolesSwitcherReducer should remove Ansible role from assigned 1`] = `
Object {
  "assignedPagination": Object {
    "page": 1,
    "perPage": 20,
  },
  "assignedRoles": Array [
    Object {
      "id": 4,
      "inherited": true,
      "name": "geerlingguy.java",
    },
    Object {
      "id": 5,
      "name": "naftulikay.golang",
    },
  ],
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "itemCount": 12,
  "loading": false,
  "pagination": Object {
    "page": 1,
    "perPage": 5,
  },
  "results": Array [
    Object {
      "id": 1,
      "name": "sthirugn.motd",
    },
    Object {
      "id": 2,
      "name": "jtyr.ntp",
    },
    Object {
      "id": 3,
      "name": "rvm.ruby",
    },
    Object {
      "id": 4,
      "name": "geerlingguy.java",
    },
    Object {
      "id": 5,
      "name": "naftulikay.golang",
    },
    Object {
      "id": 6,
      "name": "theforeman.foreman_scap_client",
    },
    Object {
      "id": 7,
      "name": "ansible.ansible",
    },
    Object {
      "id": 8,
      "name": "puppet.puppet",
    },
    Object {
      "id": 9,
      "name": "chef.chef",
    },
    Object {
      "id": 10,
      "name": "salt.salt",
    },
    Object {
      "id": 11,
      "name": "anonymous.something",
    },
    Object {
      "id": 6,
      "name": "theforeman.foreman_scap_client",
    },
  ],
}
`;

exports[`AnsibleRolesSwitcherReducer should return initial state 1`] = `
Object {
  "assignedPagination": Object {
    "page": 1,
    "perPage": 10,
  },
  "assignedRoles": Array [],
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "inheritedRoleIds": Array [],
  "itemCount": 0,
  "loading": false,
  "pagination": Object {
    "page": 1,
    "perPage": 10,
  },
  "results": Array [],
}
`;

exports[`AnsibleRolesSwitcherReducer should start loading on Ansible roles request 1`] = `
Object {
  "assignedPagination": Object {
    "page": 1,
    "perPage": 10,
  },
  "assignedRoles": Array [],
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "inheritedRoleIds": Array [],
  "itemCount": 0,
  "loading": true,
  "pagination": Object {
    "page": 1,
    "perPage": 10,
  },
  "results": Array [],
}
`;

exports[`AnsibleRolesSwitcherReducer should stop loading on Ansible roles error 1`] = `
Object {
  "assignedPagination": Object {
    "page": 1,
    "perPage": 10,
  },
  "assignedRoles": Array [],
  "error": Object {
    "errorMsg": "Failed to fetch Ansible Roles from server.",
    "statusText": "500",
  },
  "inheritedRoleIds": Array [],
  "itemCount": 0,
  "loading": false,
  "pagination": Object {
    "page": 1,
    "perPage": 10,
  },
  "results": Array [],
}
`;

exports[`AnsibleRolesSwitcherReducer should stop loading on Ansible roles success 1`] = `
Object {
  "assignedPagination": Object {
    "page": 1,
    "perPage": 10,
  },
  "assignedRoles": Array [
    Object {
      "id": 4,
      "name": "geerlingguy.java",
    },
    Object {
      "id": 5,
      "name": "naftulikay.golang",
    },
    Object {
      "id": 6,
      "name": "theforeman.foreman_scap_client",
    },
  ],
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "inheritedRoleIds": Array [
    4,
  ],
  "itemCount": 11,
  "loading": false,
  "pagination": Object {
    "page": 1,
    "perPage": 5,
  },
  "results": Array [
    Object {
      "id": 1,
      "name": "sthirugn.motd",
    },
    Object {
      "id": 2,
      "name": "jtyr.ntp",
    },
    Object {
      "id": 3,
      "name": "rvm.ruby",
    },
    Object {
      "id": 4,
      "name": "geerlingguy.java",
    },
    Object {
      "id": 5,
      "name": "naftulikay.golang",
    },
    Object {
      "id": 6,
      "name": "theforeman.foreman_scap_client",
    },
    Object {
      "id": 7,
      "name": "ansible.ansible",
    },
    Object {
      "id": 8,
      "name": "puppet.puppet",
    },
    Object {
      "id": 9,
      "name": "chef.chef",
    },
    Object {
      "id": 10,
      "name": "salt.salt",
    },
    Object {
      "id": 11,
      "name": "anonymous.something",
    },
  ],
}
`;
